river/Cursor.zig

```diff
@@ -98,17 +98,15 @@ pub const ResizeDirection = enum {
 
     /// Converts wlr.Edges into ResizeDirection, falling back to .botright when
     /// in case of nonsense values.
-    pub fn fromWlrEdges(edges: wlr.Edges) ResizeDirection {
+    pub fn fromWlrEdges(edges: wlr.Edges) !ResizeDirection {
         // Only two edges may be active at max and they must not be opposing.
-        // However a client can technically request these nonsense modes. In
-        // that case, we just fall back the bottom right corner.
-        if (edges.top and edges.top == edges.bottom) return .botright;
-        if (edges.right and edges.right == edges.left) return .botright;
-        if (edges.top and edges.right and edges.bottom and edges.left) return .botright;
-        if (edges.top and edges.right and edges.bottom) return .botright;
-        if (edges.top and edges.right and edges.left) return .botright;
-        if (edges.top and edges.bottom and edges.left) return .botright;
-        if (edges.right and edges.bottom and edges.left) return .botright;
+        if (edges.top and edges.top == edges.bottom) return error.InvalidEdges;
+        if (edges.right and edges.right == edges.left) return error.InvalidEdges;
+        if (edges.top and edges.right and edges.bottom and edges.left) return error.InvalidEdges;
+        if (edges.top and edges.right and edges.bottom) return error.InvalidEdges;
+        if (edges.top and edges.right and edges.left) return error.InvalidEdges;
+        if (edges.top and edges.bottom and edges.left) return error.InvalidEdges;
+        if (edges.right and edges.bottom and edges.left) return error.InvalidEdges;
```

Comment:
I'm pretty sure this can be simplified to just
```suggestion
        if (edges.top and edges.bottom) return error.InvalidEdges;
        if (edges.right and edges.left) return error.InvalidEdges;
```